circuit-lib/light-prover-client/src/mock_batched_forester.rs

```diff
@@ -0,0 +1,181 @@
+use std::fmt::Error;
+
+use light_hasher::{Hasher, Poseidon};
+use light_merkle_tree_reference::MerkleTree;
+use light_utils::bigint::bigint_to_be_bytes_array;
+use reqwest::Client;
+
+use crate::{
+    batch_append::{calculate_hash_chain, get_batch_append_inputs},
+    batch_update::get_batch_update_inputs,
+    gnark::{
+        batch_append_json_formatter::append_inputs_string,
+        batch_update_json_formatter::update_inputs_string,
+        constants::{PROVE_PATH, SERVER_ADDRESS},
+        proof_helpers::{compress_proof, deserialize_gnark_proof_json, proof_from_json_struct},
+    },
+};
+
+// TODO: rename to MockBatchedForester
+pub struct MockBatchedForester<const HEIGHT: usize> {
```

Comment:
nit: since we don't have any arrays which depend on `HEIGHT` and we have only dynamic vectors, perhaps we can skip const generic?

In retrospect, I think that introducing them in the previous Merkle trees was a mistake - it prevented IDL generation and made code messier.